website/source/docs/auth/aws-iam.html.md

```diff
@@ -0,0 +1,606 @@
+---
+layout: "docs"
+page_title: "Auth Backend: AWS-IAM"
+sidebar_current: "docs-auth-aws-iam"
+description: |-
+  The aws-iam backend allows automated authentication AWS IAM principals.
+---
+
+# Auth Backend: aws-iam
+
+The aws-iam auth backend provides a secure introduction mechanism for AWS IAM
+principals. This allows you to reduce the problem of securely introducing a
+Vault token to the problem of securely introducing AWS IAM credentials, which
+AWS has already solved in a number of use cases, such as via EC2 Instance
+Profiles and IAM roles attached to Lambda functions. It also allows you to have
+a consistent workflow between developers working on a local laptop with tools
+such as Hologram.
+
+## Comparison with AWS-EC2 Authentication Backend
+
+The AWS-IAM and AWS-EC2 authentication backends serve similar purposes. Both
```

Comment:
@joelthompson I agree with your statement that launching instance into an instance profile is treated as a much more security sensitive operation than VPC/subnet/etc.

My concern here / use case:

1. Limiting blast radius. Most of our "real" apps are deployed at least into multiple subnets, if not multiple VPCs. I'd like to be able to lock down to just one of those.
2. We don't have "EC2 Admins"; our development teams are self-service, and devs have a team-based IAM role that lets them manage their own infrastructure... in the best (from a security stance, and something that we've only partially done so far) case, team members are assigned to an IAM Role that allows them to deploy into a given VPC, and possibly a given set of subnets. So in this case, we'd get a real benefit from that.
3. Security through obscurity. Even if a user had (as some do....) nearly-unlimited privileges on their IAM user, and their creds got compromised, knowing that getting FooApp's Vault access requires being in a specific subnet in a specific VPC is probably less intuitive for an attacker to figure out than "let me spin up an instance with the same IAM Role." Yeah... I know... that's a bit of an awful statement...